### PR TITLE
Lexer hint

### LINKED ISSUE #3735: More helpful error message on `text*//`
### Description

No reason to fail there since we accept `text//`

### Reproduction URL

_No response_

### Operating system

_No response_

### Typst version

- [X] I am using the latest version of Typst

### PR DESCRIPTION
**Pull request by astrale-sharp**
Closes #3735

Hey there, long time no see.

First commit adds hint support for the Lexer, second one adds a message discussed in 3735.

### CODE REVIEW

File: crates/typst-syntax/src/node.rs [hunk 1]
```diff
@@ -674,6 +667,14 @@ pub struct SyntaxError {
 }
 
 impl SyntaxError {
+    pub fn new(message: impl Into<EcoString>) -> Self {
```
[the remark this answers is not in the document]
Comment (astrale-sharp) [reply crates/typst-syntax/src/node.rs:671]: I did, it's very brief, should I add that it doesn't contain any hints? (feels unnecessary)

**Comment by astrale-sharp** [comment]
I just need to fix the tests now

**Comment by laurmaedje** [comment]
Thanks!

**Comment by astrale-sharp** [comment]
What in the github is happening here :rofl:

**Comment by laurmaedje** [comment]
GitHub actions were down so I couldn't get the merge queue to do it's thing ^^

**Pull request merged**